docker/machine.go

```diff
@@ -0,0 +1,228 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package docker
+
+import (
+	"encoding/base64"
+	"fmt"
+	"strings"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	"sigs.k8s.io/cluster-api-provider-docker/cloudinit"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha2"
+	"sigs.k8s.io/cluster-api/util"
+	"sigs.k8s.io/kind/pkg/cluster/constants"
+	"sigs.k8s.io/kind/pkg/cluster/nodes"
+)
+
+const (
+	defaultImage = "kindest/node:v1.15.0"
+)
+
+// Machine implement a service for managing the docker containers hosting a kubernetes nodes.
+type Machine struct {
```

Comment:
What if the nodes.Node was a field on Machine?

We could look it up at NewMachine time and then reuse the reference. Same with LB. That way, each `docker.Machine` would know what container it's referencing and that logic falls out of the Reconciler.

Reply:
done